src/pages/spawn/spawnVolume/spawnVolumeTable/spawnVolumeTableActions/migrateBtn/MigrateVolumeModal.tsx

```diff
@@ -0,0 +1,166 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { useMutation } from "@apollo/client";
+import { Body } from "@leafygreen-ui/typography";
+import { useSpawnAnalytics } from "analytics";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import {
+  formToGql,
+  getFormSchema,
+  useLoadFormSchemaData,
+  useVirtualWorkstationDefaultExpiration,
+  validateSpawnHostForm,
+  FormState,
+} from "components/Spawn/spawnHostModal";
+import { SpruceForm } from "components/SpruceForm";
+import { useToastContext } from "context/toast";
+import {
+  MigrateVolumeMutation,
+  MigrateVolumeMutationVariables,
+} from "gql/generated/types";
+import { MIGRATE_VOLUME } from "gql/mutations";
+import { MyVolume } from "types/spawn";
+import { omit } from "utils/object";
+
+interface MigrateVolumeModalProps {
+  volume: MyVolume;
+  open: boolean;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const MigrateVolumeModal: React.VFC<MigrateVolumeModalProps> = ({
+  open,
+  setOpen,
+  volume,
+}) => {
+  const [submitClickCount, setSubmitClickCount] = useState(0);
```

Comment:
I'm not a big fan of this pattern.  I think it might make sense to use something like `useReducer` and treat this more like a state machine instead of just counting the number of times the user clicked the submit button.

`useReducer` would be more verbose about what is actually happening to the modal state and would make the code more understandable since you would be able to correspond a dispatched action to the state of the modal instead of relying on an index.